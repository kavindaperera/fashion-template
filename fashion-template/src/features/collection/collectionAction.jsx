import { toastr } from 'react-redux-toastr'
import { FETCH_PRODUCTS, FETCH_SUBITEMS , GET_CURRENCY }  from "./collectionConstants";
import {asyncActionStart,asyncActionFinish,asyncActionError
} from "../async/asyncActions";
import firebase from '../../app/config/firebase'


/*export const getProducts = (storeId) => 
  async (dispatch, getState) => {
    let store = storeId;
    const firestore = firebase.firestore();
    const productsQuery = firestore.collection('products').where('store','==',store);
    try {
      dispatch(asyncActionStart());
      let querySnap = await productsQuery.get()
      let products = [];
      for (let i=0; i < querySnap.docs.length; i++) {
        let product = {...querySnap.docs[i].data(), id: querySnap.docs[i].id};
        products.push(product);
      }
      console.log("action",products)
      dispatch({type: FETCH_PRODUCTS, payload: {products}})
      dispatch(asyncActionFinish());
    } catch (error) {
      console.log("ERROR_ERROR_ERROR_ERROR",error)
      dispatch(asyncActionError())
    }
  }*/

  /*export const getSubItems = (itemId,storeId) =>
    async (dispatch, getState) => {
      dispatch(asyncActionStart());
      const firestore = firebase.firestore();
      console.log('ids',itemId,storeId)
      const subItems = [];
      const itemQuery = firestore.collection('Stores').doc(storeId).collection('Items').doc(itemId);

      try{
        dispatch(asyncActionStart());
        let itemQuerySnap = await itemQuery.get()
        const subItemQuery = firestore.collection('Stores').doc(storeId).collection('SubItems').where('item','==',itemQuery);
        let subItemQuerySnap = await subItemQuery.get()
        subItemQuerySnap.forEach(doc => {
          let subItem = doc.data();
          if (subItem.item){subItem.item.get().then(res => {
            subItem.item = res.data();
            subItems.push(subItem);
          })}
        })
        dispatch({type: FETCH_SUBITEMS, payload: {subItems}})
        dispatch(asyncActionFinish());

      } catch (error){
        console.log("ERROR_ERROR_ERROR_ERROR",error)
        dispatch(asyncActionError())
      }
  }*/


/*export const getSubItems = (item,store) =>
  async (dispatch, getState) => {
    try{
    dispatch(asyncActionStart());
    const firestore = firebase.firestore();
    const subItemRefs = item.subItems;
    const subItems = [];
    const subItemIds = [];
    subItemRefs.map(subItem => {
      subItemIds.push(subItem.id)
  })
    subItemIds.map(id => {
      const subItemQuery = firestore.collection('Stores').doc(store).collection('SubItems').doc(id);
      let getDoc = subItemQuery.get()
      .then(doc => {
        subItems.push(doc.data());
    })
  })
  console.log(subItems)
  dispatch({type: FETCH_SUBITEMS, payload: {subItems}})
  dispatch(asyncActionFinish());

}catch(error){
  console.log("ERROR_ERROR_ERROR_ERROR",error)
        dispatch(asyncActionError())
}
}*/


export const getCurrency = (config,store) =>
  async (dispatch, getState) => {
  try{
    dispatch(asyncActionStart());
    let symbol = 'X';
    const currencies = config.currencies;
    const storeCurrency = store.currency;
    var value;
    Object.keys(currencies).forEach(function(key) {
      value = currencies[key];
      if (key==storeCurrency){ symbol=value}
    });

    dispatch({type: GET_CURRENCY, payload: {symbol}})
    dispatch(asyncActionFinish());
  } catch(error){
        console.log("ERROR_Currency",error)
        dispatch(asyncActionError())
  }
}

