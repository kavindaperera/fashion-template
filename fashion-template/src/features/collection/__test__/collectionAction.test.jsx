import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk'
// Actions to be tested
import * as selectActions from '../collectionAction';


const middlewares = [thunk] // add your middlewares like `redux-thunk`
const mockStore = configureStore(middlewares);
const store = mockStore();

const  config ={
    "currencies": {
      "AUD": "$",
      "CAD": "$",
      "CNY": "¥",
      "EUR": "€",
      "GBP": "£",
      "HKD": "$",
      "JPY": "¥",
      "LKR": "Rs.",
      "RUB": "р.",
      "USD": "$"
    },
    "orderStates": [
      "Paid",
      "Shipped",
      "Delivered"
    ]
  }

const store_1 ={
    currency: 'USD'
}

const store_2 ={
    currency: 'EUR'
}

describe('getCurrency', () => {
    beforeEach(() => { // Runs before each test in the suite
        store.clearActions();
      });
    test('Dispatches the correct action and payload', () => {
      const expectedActions = [
                    {
                    "type": "ASYNC_ACTION_START",
                    },
                    {
                    'type': 'GET_CURRENCY',
                    'payload':{ symbol: '$'}
                    },
                    {
                    "type": "ASYNC_ACTION_FINISH",
                    },
                    ];
      store.dispatch(selectActions.getCurrency(config,store_1));
      expect(store.getActions()).toEqual(expectedActions);
    });
  });

describe('getCurrency', () => {
    beforeEach(() => { // Runs before each test in the suite
        store.clearActions();
      });
    test('Dispatches the correct action and payload', () => {
      const expectedActions = [
                    {
                    "type": "ASYNC_ACTION_START",
                    },
                    {
                    'type': 'GET_CURRENCY',
                    'payload':{ symbol: '€'}
                    },
                    {
                    "type": "ASYNC_ACTION_FINISH",
                    },
                    ];
      store.dispatch(selectActions.getCurrency(config,store_2));
      expect(store.getActions()).toEqual(expectedActions);
    });
  });

describe('getCurrency', () => {
    beforeEach(() => { // Runs before each test in the suite
        store.clearActions();
      });
    test('Dispatches the correct action and payload', () => {
      store.dispatch(selectActions.getCurrency(config,store_1));
      expect(store.getActions()).toMatchSnapshot();
    });
  });

describe('getCurrency', () => {
    beforeEach(() => { // Runs before each test in the suite
        store.clearActions();
      });
    test('Dispatches the correct action and payload', () => {
      store.dispatch(selectActions.getCurrency(config,store_2));
      expect(store.getActions()).toMatchSnapshot();
    });
  });