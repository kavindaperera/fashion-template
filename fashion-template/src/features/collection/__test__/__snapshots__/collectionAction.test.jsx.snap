// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getCurrency Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "type": "ASYNC_ACTION_START",
  },
  Object {
    "payload": Object {
      "symbol": "$",
    },
    "type": "GET_CURRENCY",
  },
  Object {
    "type": "ASYNC_ACTION_FINISH",
  },
]
`;

exports[`getCurrency Dispatches the correct action and payload 2`] = `
Array [
  Object {
    "type": "ASYNC_ACTION_START",
  },
  Object {
    "payload": Object {
      "symbol": "â‚¬",
    },
    "type": "GET_CURRENCY",
  },
  Object {
    "type": "ASYNC_ACTION_FINISH",
  },
]
`;
